### PR TITLE
CPR-315 Map initial NOMIS domain event to CPR person

### PR DESCRIPTION
**Pull request by H-Iwanejko**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/model/prisoner/Events.kt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package uk.gov.justice.digital.hmpps.personrecord.client.model.prisoner
+
+object Events {
+  const val PRISONER_CREATED = "prisoner-offender-search.prisoner.created"
+  const val PRISONER_UPDATED = "prisoner-offender-search.prisoner.updated"
+}
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/client/model/prisoner/Events.kt:6]: These look like test only constants - if so they would be better in the `TEST` tree. I also don't think we need to use an `object` to store strings - they can just be defined in a file

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/model/types/SQSMessageType.kt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package uk.gov.justice.digital.hmpps.personrecord.model.types
+
+object SQSMessageType {
+  const val NOTIFICATION = "Notification"
+}
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/model/types/SQSMessageType.kt:5]: no need for an object

File: src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/type/TelemetryEventType.kt [hunk 3]
```diff
@@ -4,8 +4,6 @@ enum class TelemetryEventType(val eventName: String) {
   HMCTS_MESSAGE_RECEIVED("CprHMCTSMessageReceived"),
   HMCTS_PROCESSING_FAILURE("CprHMCTSMessageProcessingFailed"),
   MATCH_CALL_FAILED("CprMatchCallFailed"),
-  NOMIS_CREATE_MESSAGE_RECEIVED("CprNOMISCreateMessageReceived"),
-  NOMIS_UPDATE_MESSAGE_RECEIVED("CprNOMISUpdateMessageReceived"),
```
Comment (markreesmoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/type/TelemetryEventType.kt:8]: 👍

File: src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/integration/MessagingMultiNodeTestBase.kt [hunk 4]
```diff
@@ -68,7 +68,11 @@ abstract class MessagingMultiNodeTestBase : IntegrationTestBase() {
       val allEvents = telemetryRepository.findAllByEvent(event.eventName)
       val matchingEvents = allEvents?.filter {
         expected.entries.map { (k, v) ->
-          JSONObject(it.properties).get(k).equals(v)
+          val jsonObject = JSONObject(it.properties)
+          when {
+            (jsonObject.has(k)) -> jsonObject.get(k).equals(v)
+            else -> false
+          }
```
Comment (markreesmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/integration/MessagingMultiNodeTestBase.kt:75]: What's this for?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/test/messages/PrisonerSearchMessages.kt [hunk 5]
```diff
@@ -0,0 +1,165 @@
+package uk.gov.justice.digital.hmpps.personrecord.test.messages
+
+fun prisonerSearchResponse(nomsNumber: String) = """
```
Comment (markreesmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/test/messages/PrisonerSearchMessages.kt:3]: 👍 although I think we will find that passing in an object with properties including nomsNumber becomes more useful as we add tests

File: src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/integration/MessagingMultiNodeTestBase.kt (hunk 4 above)
Comment (H-Iwanejko) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/personrecord/integration/MessagingMultiNodeTestBase.kt:75]: This fixes an issue if you have 2 separate events with different maps. If you search for a property that don't exist on the event it throws a JSON exception. This way it checks all regardless of property map

**Review by markreesmoj — APPROVED** [review]
👍

**Pull request merged**